File: complete_backlog/app/controllers/Application.java
package controllers;

import play.mvc.*;
import java.util.*;
import models.*;

public class Application extends Controller {

    public static void index() {
      List<Story> onDeck = Story.find("byPhase", "onDeck").fetch();
      List<Story> inProgress = Story.find("byPhase", "inProgress").fetch();
      List<Story> done = Story.find("byPhase", "done").fetch();
      render(onDeck, inProgress, done);
    }
    
    public static void createStory(String title) {
      Story s = new Story(title, "onDeck").save();
      renderJSON(s);
    }
    
    public static void changeStatus(Long storyId) {
      Story story = Story.findById(storyId);
      story.moveToNextPhase();
      index();
    }
}

File: 1devday/play-scala/story-backlog/app/application.scala
package controllers
 
import play._
import play.mvc._
import models._
 
object Application extends Controller {
  def index = {
    val stories = Stories.find("order by id desc").fetch()
    render(stories)
  }
  
  def createStory(title: String) = {
    val story = new Story(title, "deck")
    renderJSON(story.save())
  }
}